test-it/karma/fixtures/javascriptinvoke/CloseReference.via

```diff
@@ -0,0 +1,111 @@
+define(DynamicReferencesVI dv(VirtualInstrument (
+    Locals: c(
+        ce(dv(.ErrorCluster (true 100 'error' )) errorIn)
+        e(c(e(.Boolean status) e(.Int32 code) e(.String source)) err)
+        e(c(e(.Boolean status) e(.Int32 code) e(.String source)) getObjectReferenceError1)
+        e(c(e(.Boolean status) e(.Int32 code) e(.String source)) getObjectReferenceError2)
+
+        e(c(e(.Boolean status) e(.Int32 code) e(.String source)) noErrorInAndValidReferenceError)
+        e(c(e(.Boolean status) e(.Int32 code) e(.String source)) errorInAndValidReferenceError)
+        e(c(e(.Boolean status) e(.Int32 code) e(.String source)) noErrorInAndInvalidReferenceError)
+        e(c(e(.Boolean status) e(.Int32 code) e(.String source)) errorInAndInvalidReferenceError)
+
+        e(.JavaScriptDynamicRefNum objectReference)
+        e(dv(.Boolean false) isNotAValidRefnum1)
+        e(dv(.Boolean false) isNotAValidRefnum2)
+        e(dv(.Boolean false) isNotAValidRefnum3)
+        e(dv(.Boolean false) isNotAValidRefnum4)
+        e(.Occurrence occurrence)
+    )
+    clump (
+        // no error in, valid reference, close reference => no error out, reference closed
+        JavaScriptInvoke(occurrence false getObjectReferenceError1 'NI_GetObjectFunction' objectReference)
+        CloseReference(objectReference noErrorInAndValidReferenceError)
```

Comment:
Can you add IsNotANumPathRefnum checks before CloseReference so we can be sure the reference was valid before it was closed?

Reply:
Done